modulefiles/obsproc_common.lua

```diff
@@ -2,17 +2,19 @@ help([[
 Load common modules to build obsproc on all machines
 ]])
 
-hdf5_ver=os.getenv("hdf5_ver") or "default"
-netcdf_ver=os.getenv("netcdf_ver") or "default"
-bacio_ver=os.getenv("bacio_ver") or "default"
-w3emc_ver=os.getenv("w3emc_ver") or "default"
-sp_ver=os.getenv("sp_ver") or "default"
-sigio_ver=os.getenv("sigio_ver") or "default"
-nemsio_ver=os.getenv("nemsio_ver") or "default"
-bufr_ver=os.getenv("bufr_ver") or "default"
+hdf5_ver=os.getenv("hdf5_ver") or "None"
+netcdf_c_ver=os.getenv("netcdf_c_ver") or "None"
+netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "None"
+bacio_ver=os.getenv("bacio_ver") or "None"
+w3emc_ver=os.getenv("w3emc_ver") or "None"
+sp_ver=os.getenv("sp_ver") or "None"
+sigio_ver=os.getenv("sigio_ver") or "None"
+nemsio_ver=os.getenv("nemsio_ver") or "None"
+bufr_ver=os.getenv("bufr_ver") or "None"
 
 load(pathJoin("hdf5", hdf5_ver))
-load(pathJoin("netcdf", netcdf_ver))
+load(pathJoin("netcdf-c", netcdf_c_ver))
```

Comment:
Is "netcdf-c" the same as "netcdf-C"?

Reply:
@AlexanderRichert-NOAA can you answer Iliana's question?

@ilianagenkova here is the `module show` for the `netcdf-c/4.9.2` (on Jet):
```
[Kate.Friedman@fe6 v1.1.3]$ module show netcdf-c/4.9.2
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
   /misc/contrib/spack-stack/spack-stack-1.6.0/envs/gsi-addon-intel/install/modulefiles/intel-oneapi-mpi/2021.5.1/intel/2021.5.0/netcdf-c/4.9.2.lua:
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
whatis("Name : netcdf-c")
whatis("Version : 4.9.2")
whatis("Target : core2")
whatis("Short description : NetCDF (network Common Data Form) is a set of software libraries and machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data. This is the C distribution.")
help([[Name   : netcdf-c]])
help([[Version: 4.9.2]])
help([[Target : core2]])
help()
help([[NetCDF (network Common Data Form) is a set of software libraries and
machine-independent data formats that support the creation, access, and
sharing of array-oriented scientific data. This is the C distribution.]])
depends_on("c-blosc/1.21.5")
depends_on("curl/8.4.0")
depends_on("hdf5/1.14.0")
depends_on("zlib/1.2.13")
depends_on("zstd/1.5.2")
prepend_path("PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/bin")
prepend_path("MANPATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/share/man")
prepend_path("LD_LIBRARY_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/lib")
prepend_path("DYLD_LIBRARY_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/lib")
prepend_path("CPATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/include")
prepend_path("PKG_CONFIG_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/lib/pkgconfig")
prepend_path("CMAKE_PREFIX_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/.")
prepend_path("PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/bin")
prepend_path("MANPATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/share/man")
prepend_path("PKG_CONFIG_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/lib/pkgconfig")
prepend_path("CMAKE_PREFIX_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/.")
append_path("HDF5_PLUGIN_PATH","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24/plugins")
setenv("netcdf_c_ROOT","/misc/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/netcdf-c-4.9.2-wxvro24")
append_path("MANPATH","")
```

Reply:
@KateFriedman-NOAA @ilianagenkova netcdf-c is the C-language netCDF library.  Spack-stack installs netcdf-c and netcdf-fortran separately to allow for increased flexibility.

This is different than the `netcdf-C` module on WCOSS2, which contains both the C- and Fortran-language libraries.  Moreover, the `-A`, `-B`, and `-C` libraries on WCOSS2 are tied to three HPC-stack co-installations of a handful of packages (HDF5, netCDF, ESMF, and pnetcdf) and should be used together (e.g. hdf5-A should be loaded with netcdf-A and esmf-A).

Reply:
Glad I asked! Thanks for the explanation, @DavidHuber-NOAA

Reply:
Good to know, thanks @DavidHuber-NOAA !